src/App.tsx

```diff
@@ -15,48 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+
+  const hadnleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const handleSort = (newSort: SortType) => setSortType(newSort);
+
+  const handleReset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  useEffect(() => {
+    const reorderGoods = () => {
+      const goodsCopy = [...goodsFromServer];
+
+      if (sortType === SortType.ALPHABET) {
+        goodsCopy.sort((a, b) => a.localeCompare(b));
+      } else if (sortType === SortType.LENGTH) {
+        goodsCopy.sort((a, b) => a.length - b.length);
+      }
+
+      if (isReversed) {
+        goodsCopy.reverse();
+      }
+
+      setVisibleGoods(goodsCopy);
+    };
+
+    reorderGoods();
+  }, [sortType, isReversed]);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${
+            sortType === SortType.ALPHABET ? '' : 'is-light'
+          }`}
```

Comment:
it is better to use libraries like classnames or clsx for conditional classNames